marlowe-playground-client/src/Projects/Types.purs

```diff
@@ -0,0 +1,45 @@
+module Projects.Types where
+
+import Prelude
+import Analytics (class IsEvent, Event)
+import Analytics as A
+import Data.Generic.Rep (class Generic)
+import Data.Generic.Rep.Show (genericShow)
+import Data.Lens (Lens')
+import Data.Lens.Record (prop)
+import Data.Maybe (Maybe(..))
+import Data.Symbol (SProxy(..))
+import Gist (Gist, GistId)
+import Network.RemoteData (RemoteData(..))
+
+data Lang
+  = Marlowe
+  | Haskell
+  | Javascript
+  | Blockly
+
+derive instance genericLang :: Generic Lang _
+
+instance showLang :: Show Lang where
+  show lang = genericShow lang
+
+data Action
+  = LoadProjects
+  | LoadProject Lang GistId
+
+defaultEvent :: String -> Event
+defaultEvent s = A.defaultEvent $ "Projects." <> s
```

Comment:
This would be much better if you arranged it such that `{ action = s, category = Just "Projects" }`.

Also dropping the full stop.

Reply:
oh didn't look at all those other fields!